ai_engine.py

```diff
@@ -40,7 +40,7 @@ def minimax_white(self, game_state, depth, alpha, beta, maximizing_player, playe
             max_evaluation = -10000000
             all_possible_moves = game_state.get_all_legal_moves("black")
             for move_pair in all_possible_moves:
-                game_state.move_piece(move_pair[0], move_pair[1], True)
+                game_state.move_piece(move_pair[0], move_pair[1], True, is_fake=True)
```

Comment:
You didn't solve the issue of the evalutaion, try to find it

Reply:
Are you referring to the bug that the bot plays unwisely?
If so, please see the first commit above ("3 bug fixes").